tools/use-new-urls.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+
+__requires__ = [
+    "boto3",
+    "click == 7.*",
+    "click-loglevel ~= 0.2",
+    "dandi >= 0.14.0",
+    "datalad",
+]
+
+import logging
+from pathlib import Path
+import sys
+from urllib.parse import urlparse, urlunparse
+
+import boto3
+from botocore import UNSIGNED
+from botocore.client import Config
+import click
+from click_loglevel import LogLevel
+from dandi.dandiapi import DandiAPIClient
+from datalad.api import Dataset
+
+log = logging.getLogger(Path(sys.argv[0]).name)
+
+
+@click.command()
+@click.option(
+    "-l",
+    "--log-level",
+    type=LogLevel(),
+    default="INFO",
+    help="Set logging level",
+    show_default=True,
+)
+@click.argument("datasets_path", type=click.Path(exists=True, file_okay=False))
+@click.argument("dandisets", nargs=-1)
+def main(datasets_path, dandisets, log_level):
+    logging.basicConfig(
+        format="%(asctime)s [%(levelname)-8s] %(name)s %(message)s",
+        datefmt="%Y-%m-%dT%H:%M:%S%z",
+        level=log_level,
+        force=True,  # Override dandi's settings
+    )
+    URLUpdater(Path(datasets_path)).run(dandisets)
+
+
+class URLUpdater:
+    def __init__(self, datasets_path: Path):
+        self.datasets_path = datasets_path
+        self.dandi_client = DandiAPIClient("https://api.dandiarchive.org/api")
+        self.s3client = boto3.client("s3", config=Config(signature_version=UNSIGNED))
+
+    def run(self, dandisets=()):
+        with self.dandi_client.session():
+            for did in dandisets or self.get_dandiset_ids():
+                dsdir = self.datasets_path / did
+                log.info("Updating URLs for Dandiset %s", did)
+                ds = Dataset(str(dsdir))
+                self.update_dandiset_urls(did, ds)
+                log.info("Pushing to sibling")
+                ds.push(to="github")
+
+    def update_dandiset_urls(self, dandiset_id, ds):
+        if ds.repo.dirty:
+            raise RuntimeError("Dirty repository; clean or save before running")
+        dsdir = ds.pathobj
+        for a in self.dandi_client.get_dandiset_assets(
+            dandiset_id, "draft", include_metadata=False
+        ):
+            path = a["path"]
+            log.info("Processing asset %s", path)
+            if ds.repo.is_under_annex(path, batch=True):
+                for url in ds.repo.get_urls(path, batch=True):
+                    log.info("Removing URL %s from asset", url)
+                    ds.repo.call_annex(
+                        [
+                            "rmurl",
+                            "-c",
+                            "annex.alwayscommit=false",
```

Comment:
I wonder if that is what forced annex to do the commit per file -- may be `-c` should be provided at `git` command invocation level... if you use `_call_annex` instead of `call_annex` you could pass `git_options=["-c", "annex.alwayscommit=false"]`.